### PR TITLE
add translator service

### PR DESCRIPTION
**Pull request by sensorario**
| Question | Answer |
| --- | --- |
| branch | master |
| bug fix | no |
| new feature | yes |
| BC breaks | no |
| deprecations | no |
| test pass | no |

### CODE REVIEW

File: Services/Translator.php [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    {
+        if (!isset($this->locale)) {
+            throw new \RuntimeException(
+                'Undefined locale'
+            );
+        }
+
+        if (!$entity->hasTranslations()) {
+            throw new \RuntimeException(
+                'Missing "' . $this->locale . '" translation'
```
Comment (sensorario) [inline Services/Translator.php]: Ho risolto ...

Comment (samuele-mrapps) [inline Services/Translator.php]: Ottimo! 👍

**Review by sensorario — COMMENTED** [review]
Comment summary ...

File: Entity/TranslatedEntity.php [hunk 2]
```diff
@@ -0,0 +1,19 @@
+<?php
+
+namespace Mrapps\BackendBundle\Entity;
+
+use Doctrine\ORM\Mapping as ORM;
+
+abstract class TranslatedEntity
```
Comment (samuele-mrapps) [inline Entity/TranslatedEntity.php]: test

**Review by samuele-mrapps — APPROVED** [review]

File: Services/Translator.php [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+use Mrapps\BackendBundle\Entity\TranslatedEntity;
+
+class Translator
+{
+    private $locale;
+
+    private $manager;
+
+    public function __construct(
+        EntityManager $manager
```
Comment (sensorario) [inline Services/Translator.php]: asdfadsfads

File: Services/Translator.php [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    ) {
+        $this->manager = $manager;
+    }
+
+    public function setLocale($locale)
+    {
+        $this->locale = $locale;
+
+        $this->language = $this->manager
+            ->getRepository('MrappsBackendBundle:Language')
```
Comment (sensorario) [inline Services/Translator.php]: asdfdsaf

File: Entity/TranslatedEntity.php (hunk 2 above)
Comment (sensorario) [inline Entity/TranslatedEntity.php]: tset

Comment (sensorario) [inline Entity/TranslatedEntity.php]: ok

**Pull request merged**